File: win-prog/ProgWinCS/Font Fun/RotatedFont/RotatedFont.cs
//------------------------------------------
// RotatedFont.cs © 2001 by Charles Petzold
//------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class RotatedFont: FontMenuForm
{
     const int iDegrees = 20;      // Should be divisor of 360

     public new static void Main()
     {
          Application.Run(new RotatedFont());
     }
     public RotatedFont()
     {
          Text = "Rotated Font";

          strText = "   Rotated Font";
          font = new Font("Arial", 18);
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          Brush brush = new SolidBrush(clr);
          StringFormat strfmt = new StringFormat();
          strfmt.LineAlignment = StringAlignment.Center;
          
          grfx.TranslateTransform(cx / 2, cy / 2);

          for (int i = 0; i < 360; i += iDegrees)
          {
               grfx.DrawString(strText, font, brush, 0, 0, strfmt);
               grfx.RotateTransform(iDegrees);
          }
     }
}

File: win-prog/ProgWinVB/Font Fun/RotatedFont/RotatedFont.vb
'--------------------------------------------
' RotatedFont.vb (c) 2002 by Charles Petzold
'--------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class RotatedFont
    Inherits FontMenuForm

    Const iDegrees As Integer = 20     ' Should be divisor of 360

    Shared Shadows Sub Main()
        Application.Run(New RotatedFont())
    End Sub

    Sub New()
        Text = "Rotated Font"
        strText = "   Rotated Font"
        fnt = New Font("Arial", 18)
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        Dim br As New SolidBrush(clr)
        Dim strfmt As New StringFormat()
        Dim i As Integer

        strfmt.LineAlignment = StringAlignment.Center
        grfx.TranslateTransform(cx \ 2, cy \ 2)

        For i = 0 To 359 Step iDegrees
            grfx.DrawString(strText, fnt, br, 0, 0, strfmt)
            grfx.RotateTransform(iDegrees)
        Next i
    End Sub
End Class
